src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const cloneArray = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear' :
+        Object.keys(stateClone).forEach(key => delete stateClone[key]);
+        break;
+      case 'addProperties' :
+        Object.assign(stateClone, action.extraData);
+        break;
+      case 'removeProperties' :
+        for (const removleKey of action.keysToRemove) {
+          delete stateClone[removleKey];
+        }
```

Comment:
You can use here also method `forEach()`,  but I don't think that this is a mistake.